test/integration/aggregate.test.ts

```diff
@@ -0,0 +1,313 @@
+      byDate: {
+        dateHistogram: {
+          column: 'founded_date',
+          calendarInterval: 'week',
+          aggs: {
+            topNames: {
+              topValues: {
+                column: 'name',
+                aggs: { maxRating: { max: { column: 'rating' } } }
+              }
+            }
+          }
+        }
+      }
+    });
+
+    expect(result.aggs.byDate.values).toHaveLength(3);
+    expect(result.aggs.byDate.values?.[0].$key).toBe('2021-12-27T00:00:00.000Z');
+    expect(result.aggs.byDate.values?.[0].$count).toBe(2);
+    expect(result.aggs.byDate.values?.[0].topNames.values).toHaveLength(1);
+    expect(result.aggs.byDate.values?.[0].topNames.values?.[0].$key).toBe('r1');
+    expect(result.aggs.byDate.values?.[0].topNames.values?.[0].$count).toBe(2);
+    expect(result.aggs.byDate.values?.[0].topNames.values?.[0].maxRating).toBe(2.2);
+    expect(result.aggs.byDate.values?.[1].$key).toBe('2022-01-03T00:00:00.000Z');
+    expect(result.aggs.byDate.values?.[1].$count).toBe(2);
+    expect(result.aggs.byDate.values?.[1].topNames.values).toHaveLength(1);
+    expect(result.aggs.byDate.values?.[1].topNames.values?.[0].$key).toBe('r2');
+    expect(result.aggs.byDate.values?.[1].topNames.values?.[0].$count).toBe(2);
+    expect(result.aggs.byDate.values?.[1].topNames.values?.[0].maxRating).toBe(4);
+    expect(result.aggs.byDate.values?.[2].$key).toBe('2022-01-10T00:00:00.000Z');
+    expect(result.aggs.byDate.values?.[2].$count).toBe(1);
+    expect(result.aggs.byDate.values?.[2].topNames.values).toHaveLength(1);
+    expect(result.aggs.byDate.values?.[2].topNames.values?.[0].$key).toBe('r3');
+    expect(result.aggs.byDate.values?.[2].topNames.values?.[0].$count).toBe(1);
+    expect(result.aggs.byDate.values?.[2].topNames.values?.[0].maxRating).toBe(5.3);
+  });
+
+  test('numeric histogram combined with sub-aggs', async () => {
+    const result = await xata.db.teams.aggregate({
+      byIndex: {
+        numericHistogram: {
+          column: 'index',
+          interval: 5,
+          aggs: { avgRating: { average: { column: 'rating' } } }
+        }
+      }
+    });
+
+    expect(result.aggs.byIndex.values).toHaveLength(3);
+    expect(result.aggs.byIndex.values?.[0].$key).toBe(0);
+    expect(result.aggs.byIndex.values?.[0].$count).toBe(2);
+    expect(result.aggs.byIndex.values?.[0].avgRating).toBe(3.55);
+    expect(result.aggs.byIndex.values?.[1].$key).toBe(5);
+    expect(result.aggs.byIndex.values?.[1].$count).toBe(2);
+    expect(result.aggs.byIndex.values?.[1].avgRating).toBe(1.6);
+    expect(result.aggs.byIndex.values?.[2].$key).toBe(10);
+    expect(result.aggs.byIndex.values?.[2].$count).toBe(1);
+    expect(result.aggs.byIndex.values?.[2].avgRating).toBe(5.3);
+  });
+});
+
+async function waitForSearchIndexing(): Promise<void> {
+  const { teams = [] } = await xata.search.byTable('longer');
+  if (teams.length === 0) {
+    await new Promise((resolve) => setTimeout(resolve, 2000));
```

Comment:
Maybe we could make this timeout configurable via env variable.

Reply:
It's not a timeout per se, it's a back off time in the while loop. For timeout we use the test suite one which is 2 minutes. We could make that one configurable from outside (I'll leave it for a future PR too).